src/app/lib/loop-http/index.ts

```diff
@@ -0,0 +1,126 @@
+import { stringify } from 'query-string';
+import { NetworkError } from './errors';
+import { parseLoopErrorResponse } from './utils';
+import * as T from './types';
+
+export type ApiMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+export class LoopHttpClient {
+  url: string;
+
+  constructor(url: string) {
+    // Remove trailing slash for consistency
+    this.url = url.replace(/\/$/, '');
+  }
+
+  // Public API methods
+
+  getLoopOutTerms = () => {
+    return this.request<T.GetLoopTermsResponse>('GET', `/v1/loop/out/terms`);
+  };
+
+  getLoopInTerms = () => {
+    return this.request<T.GetLoopTermsResponse>('GET', `/v1/loop/in/terms`);
+  };
+
+  getLoopOutQuote = (amt: string, conf: string) => {
+    return this.request<T.GetLoopQuoteResponse>(
+      'GET',
+      `/v1/loop/out/quote/${amt}?conf_target=${conf}`,
+      undefined,
+      {
+        miner_fee: '',
+        swap_fee: '',
+        prepay_amt: '',
+      },
+    );
+  };
+
+  /**
+   * TODO: Add confirmation target as required
+   * in future iterations for Loop
+   */
+  getLoopInQuote = (amt: string /*, conf: string*/) => {
+    return this.request<T.GetLoopQuoteResponse>(
+      'GET',
+      // `/v1/loop/out/quote/${amt}?conf_target=${conf}`,
+      `/v1/loop/out/quote/${amt}`,
+      undefined,
+      {
+        miner_fee: '',
+        swap_fee: '',
+        prepay_amt: '',
+      },
+    );
+  };
+
+  getLoopOut = (args: T.GetLoopOutArguments) => {
+    return this.request<T.GetLoopResponse, T.GetLoopOutArguments>(
+      'POST',
+      '/v1/loop/out',
+      args,
+    );
+  };
+
+  getLoopIn = (args: T.GetLoopInArguments) => {
+    return this.request<T.GetLoopResponse, T.GetLoopInArguments>(
+      'POST',
+      '/v1/loop/in',
+      args,
+    );
+  };
```

Comment:
These are POST requests, not GETs, so I think they should be renamed to `postLoop...`, `makeLoop...` or something like that. Ideally this nomenclature could be renamed everywhere, as it's a little confusing to call a function that starts with `get` that causes a payment to happen.

Reply:
removed "get" from the post requests.